### PR TITLE
esp32xx_rtc.c:  misc. fixes.

### PR DESCRIPTION
**Pull request by Ouss4**
## Summary
 - Remove a duplicated assignment.
 - Include `clock/clock.h` needed for `g_basetime`.
## Impact
ESP32 and ESP32-C3 RTC driver
## Testing
esp32-devkitc:rtc
esp32c3-devkit:rtc

### CODE REVIEW

File: arch/risc-v/src/esp32c3/esp32c3_rtc.c [hunk 1]
```diff
@@ -2464,8 +2467,8 @@ int up_rtc_gettime(struct timespec *tp)
     }
   else
     {
-      time_us = = esp32c3_rtc_get_time_us() +
-                    esp32c3_rtc_get_boot_time();
+      time_us = esp32c3_rtc_get_time_us() +
+                esp32c3_rtc_get_boot_time();
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_rtc.c:2471]:
```suggestion
      time_us = esp32c3_rtc_get_time_us() + esp32c3_rtc_get_boot_time();
```
nit: These lines fit in a single one, how about merging them?

File: arch/xtensa/src/esp32/esp32_rtc.c [hunk 2]
```diff
@@ -2066,8 +2069,8 @@ int up_rtc_gettime(struct timespec *tp)
     }
   else
     {
-      time_us = = esp32_rtc_get_time_us() +
-                    esp32_rtc_get_boot_time();
+      time_us = esp32_rtc_get_time_us() +
+                esp32_rtc_get_boot_time();
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32/esp32_rtc.c:2073]:
```suggestion
      time_us = esp32_rtc_get_time_us() + esp32_rtc_get_boot_time();
```

**Review by gustavonihei — APPROVED** [review]
LGTM, just made two nitpick comments. In case they get fixed, I think we're safe to merge without going through all CI jobs over again, it already passed successfully before.

File: arch/risc-v/src/esp32c3/esp32c3_rtc.c (hunk 1 above)
Comment (Ouss4) [reply arch/risc-v/src/esp32c3/esp32c3_rtc.c]: Thanks fixed.

File: arch/xtensa/src/esp32/esp32_rtc.c (hunk 2 above)
Comment (Ouss4) [reply arch/xtensa/src/esp32/esp32_rtc.c]: Done. Thanks.

**Pull request merged**